### ISSUE TITLE
cli*.exe and gnu*.exe dont work on windows (launcher.c)

### ISSUE THREAD

**Issue by mstojanovic** [pypa/setuptools#1717]
cli*.exe and gnu*.exe add a shebang on windows and due to that some executables can not be launched without the python interpreter invocation. this issue was noticed during usage of pytest(.exe) that was build using setuptools (VS14.0, python 2.7.16, will prob affect python 3.7.2). i had to change launcher.c (see lines 282-285, remove shebang) and rebuild cli*.exe and gnu*.exe (with changes to msvc-build-launcher.cmd to accommodate for VS version) to fix the error (which probably isnt the ideal solution).

TL;DR: a shebang breaks executables on windows

**Comment by jaraco** [comment]
Presumably the shebang was put there for some good reason, though I don't recall at this time wouth more investigation.

However, these executables only get used if a package is installed with `easy_install`, which is deprecated (except when used internally by setup_requires). If a package is instead installed with pip (recommended), then the executables will be generated by distlib and will have better characteristics and probably aren't subject to the bad behavior you encountered.

As a result, I'm going to close this as wont-fix, but if you have a use-case where the workaround isn't suitable for you, do please provide more detail about your use-case.
